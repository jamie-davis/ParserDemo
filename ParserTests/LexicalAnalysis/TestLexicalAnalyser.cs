using Xunit;
using Parser.LexicalAnalysis;
using TestConsoleLib;
using TestConsoleLib.Testing;
using TestConsole.OutputFormatting;

namespace ParserTests.LexicalAnalysis
{
    public class TestLexicalAnalyser
    {
        [Fact]
        public void TokensAreExtracted()
        {
            //Arrange
            var testStrings = new [] {
                "100",
                "100+",
                "100 ",
                "100L",
                "+-*/",
                "()",
                "",
                null,
                "(3 *5)+  49"
            };

            var output = new Output();

            //Act
            foreach (var testString in testStrings)
            {
                var result = LexicalAnalyser.ExtractTokens(testString);

                output.WrapLine($@"Analysis of ""{testString ?? "NULL"}"":");
                output.FormatTable(result.AsReport(rep => rep
                                                    .AddColumn(r => r.TokenType, cc => cc.LeftAlign())
                                                    .AddColumn(r => r.Text, cc => {}))
                );

                output.WriteLine();
                output.WriteLine();
            }

            //Assert
            output.Report.Verify();
        }
    }
}