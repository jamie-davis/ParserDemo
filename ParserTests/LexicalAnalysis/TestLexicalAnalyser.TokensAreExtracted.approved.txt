Analysis of "100":                                                                             
Token Type     Text                                                                            
-------------- ----                                                                            
NumericLiteral 100                                                                             
                                                                                               
                                                                                               
Analysis of "100+":                                                                            
Token Type     Text                                                                            
-------------- ----                                                                            
NumericLiteral 100                                                                             
Operator       +                                                                               
                                                                                               
                                                                                               
Analysis of "100 ":                                                                            
Token Type     Text                                                                            
-------------- ----                                                                            
NumericLiteral 100                                                                             
                                                                                               
                                                                                               
Analysis of "100L":                                                                            
Token Type   Text                                                                              
------------ ----                                                                              
Error        100L                                                                              
                                                                                               
                                                                                               
Analysis of "+-*/":                                                                            
Token Type   Text                                                                              
------------ ----                                                                              
Operator     +                                                                                 
Operator     -                                                                                 
Operator     *                                                                                 
Operator     /                                                                                 
                                                                                               
                                                                                               
Analysis of "()":                                                                              
Token Type   Text                                                                              
------------ ----                                                                              
OpenParens   (                                                                                 
CloseParens  )                                                                                 
                                                                                               
                                                                                               
                                                                                               